src/test/java/bio/terra/workspace/service/datareference/DataReferenceServiceTest.java

```diff
@@ -193,6 +159,27 @@ public void testDeleteMissingDataReference() {
             dataReferenceService.deleteDataReference(workspaceId, UUID.randomUUID(), USER_REQUEST));
   }
 
+  /**
+   * Hard code serialized values to check that code changes do not break backwards compatibility of
+   * stored JSON values. If these tests fail, your change may not work with existing databases.
+   */
+  @Test
+  public void SnapshotReferenceSerializationBackwardsCompatible() throws Exception {
+    SnapshotReference snapshotReference =
+        (SnapshotReference)
+            ReferenceObject.fromJson(
+                "{\"instanceName\":\"foo\",\"snapshot\":\"bar\"}",
+                DataReferenceType.DATA_REPO_SNAPSHOT);
+    assertEquals("foo", snapshotReference.instanceName());
+    assertEquals("bar", snapshotReference.snapshot());
+  }
+
+  @Test
+  public void SnapshotReferenceDeserializationnBackwardsCompatible() throws Exception {
+    String serializedSnapshot = SnapshotReference.create("foo", "bar").toJson();
+    assertEquals("{\"instanceName\":\"foo\",\"snapshot\":\"bar\"}", serializedSnapshot);
```

Comment:
I don't think this helps with backwards compatibility. This just serves as change detection. The stored values are the json strings, not the objects themselves.

I would like to see `y = fromJson(x.toJson); assert x equals y`, which also isn't backwards compatibility, but checks that the current version can deserialize to what it started as. Good to have for anything writing its own serialize/deserialize.